### DISCUSSION TITLE
All links to the old forum are broken

### DISCUSSION THREAD

**Discussion by verhovsky** [ImageMagick/ImageMagick#2703] [Kudos and Rants]
All links to the old forum now just redirect to the main page of the old forum, here’s an example link I found on stackoverflow: http://www.imagemagick.org/discourse-server/viewtopic.php?f=1&t=16464

Or Google “image magical convert srgb” 

![9861989B-5AD0-4DFA-9846-97BFBAA8233C](https://user-images.githubusercontent.com/5687998/95645265-56ec7100-0a8b-11eb-98e0-ffea23d08358.jpeg)

the second result just redirects to https://legacy.imagemagick.org/discourse-server/. The title text of the result sounds relevant but the actual link is useless, I’d have to search for the thread again on the website.

**Comment by matanshukry** [comment]
Also note that searching doesn't work.

While the search functionallity it self works, it results the same links as google search, which has the saame behaviour - simply redirects to the main forums.

**Comment by urban-warrior** [comment]
We're working on it.  We're fixing links as time permits and reindexing the forums by the major search engine takes time, sometimes months or more.

**Reply by verhovsky** [reply]
I noticed now that the reason you can't just make the redirection not clear the query params is because all the links on the `legacy.imagemagick.org` have small hexanumerical strings in the URL

`https://legacy.imagemagick.org/discourse-server/viewtopic**08db**.html?f=3&t=37768`

and all the pages have this too

https://legacy.imagemagick.org/discourse-server/viewforum2bc0.html?f=7&start=150

and they're different for each page.

Why did this happen? Is this the software you used for archiving the forum that adds this? Can you use something better so you could you just remove them and then have the redirection not clear the query strings? Then you just need 

https://imagemagick.org/discourse-server/viewtopic.html?f=3&t=37768

to become 

https://legacy.imagemagick.org/discourse-server/viewtopic.html?f=3&t=37768

instead of having to find the `08db`. Are you doing that manually? I hope not because the url I posted has `t=16464` in it. There's no way anyone is ever going to do 16,000 threads if it takes more than 1 second per link.

> reindexing the forums by the major search engine takes time

How many other web pages link to a page [affects that page's rank](https://en.wikipedia.org/wiki/PageRank). So, since other people aren't going to go and edit all their links to match the new page (and how could they anyway, search doesn't work, they'd need to know what was at that thread, and under which topic it was, and manually click through all the pages in that topic until they find the thread), the old urls are going to be ranked higher, since more pages link to them. 

And again, this is all useless work because for some reason you're adding 16 bits of randomness to every link for no good reason.

**Comment by verhovsky** [comment]
Googling "image magical convert srgb" I now see that the forum is completely gone from search results, so problem solved I guess.

**Comment by fmw42** [comment]
The old web site is at https://legacy.imagemagick.org/discourse-server/ but is read only now. You should still be able to search on it. No new posts there.

**Reply by Cerberus-tm** [reply]
@fmw42 Hmm that link is dead for me now. Is there any way to search the old forum at the moment?

**Reply by fmw42** [reply]
I can get to that page. But I am told they are working on fixing some issues. Please hold on.

**Reply by Cerberus-tm** [reply]
> I can get to that page. But I am told they are working on fixing some issues. Please hold on.

Ah, thanks, good to know!

**Comment by urban-warrior** [comment]
All the legacy discussion links should now work again.   If not, let us know.

**Reply by Cerberus-tm** [reply]
It seems to work, thank you very much.
